File: src/main/java/com/drunkensoftball/api/roster/repo/RosterRepository.java
package com.drunkensoftball.api.roster.repo;

import com.drunkensoftball.api.roster.domain.RosterEntry;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface RosterRepository extends JpaRepository<RosterEntry, Long> {

    RosterEntry findByUuid(String uuid);

    List<RosterEntry> findByTeamId(Long teamId);
    List<RosterEntry> findByTeamId(Long teamId, Pageable pageable);
    List<RosterEntry> findByUserId(Long userId);
    List<RosterEntry> findByUserId(Long userId, Pageable pageable);


}

File: src/main/kotlin/com/drunkensoftball/api/roster/repo/RosterRepository.kt
package com.drunkensoftball.api.roster.repo


import com.drunkensoftball.api.roster.domain.RosterEntry
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface RosterRepository : JpaRepository<RosterEntry, Long> {

    fun findByUuid(uuid: String): RosterEntry

    fun findByTeamId(teamId: Long?): List<RosterEntry>
    fun findByTeamId(teamId: Long?, pageable: Pageable): List<RosterEntry>
    fun findByUserId(userId: Long?): List<RosterEntry>
    fun findByUserId(userId: Long?, pageable: Pageable): List<RosterEntry>


}
